DOC-2666: Amazon Bedrock Integration Phase 1 (#2832)
RedisLabs/redislabs-docs: content/rc/cloud-integrations/aws-marketplace/aws-bedrock.md

---
Title: Use Redis Cloud with Amazon Bedrock
LinkTitle: Amazon Bedrock 
description: Shows how to use your Redis database with Amazon Bedrock to customize foundational models.
weight: $weight
alwaysopen: false
categories: ["RC"]
aliases: 
---

[Amazon Bedrock](https://aws.amazon.com/bedrock/) is a service that allows you to securely customize foundational models (FMs) with your own data, and to use these models without having to build complex infrastructure management. With Amazon Bedrock, users can access FMs from a variety of vendors through a single API, streamlining the process of creating generative artificial intelligence (AI).

> Amazon Bedrock allows you to choose Redis Cloud as the [vector database](https://redis.com/solutions/use-cases/vector-database/) for your models. Once your database is set up and connected to Amazon Bedrock, it will import text data from an Amazon Simple Storage Service (S3) bucket into Redis Cloud and use it to extract relevant information when prompted.

Reviewer: ```suggestion
Amazon Bedrock allows you to choose Redis Cloud as the [vector database](https://redis.com/solutions/use-cases/vector-database/) for your models. After your database is set up and connected to Amazon Bedrock, it will import text data from an Amazon Simple Storage Service (S3) bucket into Redis Cloud and use it to extract relevant information when prompted.
```

> For more information about the Redis integration with Amazon Bedrock, see the [Amazon Bedrock integration blog post](https://redis.com/blog/amazon-bedrock-integration-with-redis-enterprise/).

Reviewer 2: I assume this is a future link, as it currently displays a 404.

## Set up Redis for Bedrock

You need to set up your Redis Cloud database before you can set it as the vector database in Amazon Bedrock. To do this, you need to:

> - [Create a subscription and database in Redis Cloud](#create-subscription)
> - [Enable Transport Layer Security (TLS) for the database and save the certificates](#get-certs)
> - [Store database credentials in AWS secrets manager](#store-secret)
> - [Create a vector index in your database](#create-vector-index) for Bedrock to use

Reviewer: ```suggestion
1. [Create a subscription and database in Redis Cloud](#create-subscription)
1. [Enable Transport Layer Security (TLS) for the database and save the certificates](#get-certs)
1. [Store database credentials in AWS secrets manager](#store-secret)
1. [Create a vector index in your database](#create-vector-index) for Bedrock to use
```

> Once you have set up the database, you can use the information

Reviewer: It also seems like this sentence is incomplete.
```suggestion
After you set up the database, you can use the information
```

> Once you have set up the database, you can use the information

Reviewer 2: ```suggestion
Once you have set up the database, you can use the information.
```

    - Add the **RediSearch** and **RedisJSON** advanced capabilities to your database.
    - Set the size of the database based on the amount of data that will be pulled from your Simple Storage Service (S3) [bucket](https://docs.aws.amazon.com/AmazonS3/latest/userguide/creating-buckets-s3.html). See [Find out the size of your S3 buckets](https://aws.amazon.com/blogs/storage/find-out-the-size-of-your-amazon-s3-buckets/) to find out how much training data is stored in your S3 bucket and pick the closest size from the table below. 

    | Total Size of Documents in S3 | Database size without replication | Database size with replication |
    |-------------------------------|-----------------------------------|--------------------------------|
    | 10,000 kb                     | 135 Mb                             | 270 Mb                         |
    | 100,000 kb                    | 1.35 Gb                            | 2.7 Gb                         |
    | 1,000,000 kb                  | 13.5 Gb                              | 27 Gb                          |
    | 10,000,000 kb                 | 135 Gb                             | 270 Gb                         |

>     For more information on sizing, see the [Bedrock integration blog post](https://redis.com/blog/amazon-bedrock-integration-with-redis-enterprise#right-size-your-database-for-amazon-bedrock).

Reviewer 2: Another 404

Author: Yup, the 404s are future links. I'll check them again before I publish.

### Get TLS certificates {#get-certs}

> 1. Enable [Transport Layer Security (TLS)]({{<relref "/rc/security/database-security/tls-ssl#enable-tls">}}) for your database. Make sure to check **TLS client authentication** to require client authentication. Download the client certificates before saving your changes.

Reviewer: ```suggestion
1. Enable [Transport Layer Security (TLS)]({{<relref "/rc/security/database-security/tls-ssl#enable-tls">}}) for your database. Select **TLS client authentication** to require client authentication. Download the client certificates before saving your changes.
```

1. If you do not have the Redis Cloud server certificates, [download them]({{<relref "/rc/security/database-security/tls-ssl#download-certificates">}}) from the admin console.

### Store database credentials in AWS secrets manager {#store-secret}

In the [AWS Management Console](https://console.aws.amazon.com/), use the **Services** menu to locate and select **Security, Identity, and Compliance** > **Secrets Manager**. [Create a secret](https://docs.aws.amazon.com/secretsmanager/latest/userguide/create_secret.html) with the following fields:

- `username`: Database username
- `password`: Database password
- `serverCertificate`: Contents of the [server certificate]({{<relref "/rc/security/database-security/tls-ssl#download-certificates">}}) (`redis_ca.pem`)
- `clientCertificate`: Contents of the client certificate (`redis_user.crt`)
- `clientPrivateKey`: Contents of the client private key (`redis_user_private.key`)

> Once you store this secret, you can view and copy the [Amazon Resource Name (ARN)](https://docs.aws.amazon.com/secretsmanager/latest/userguide/reference_iam-permissions.html#iam-resources) of your secret on the secret details page. 

Reviewer: ```suggestion
After you store this secret, you can view and copy the [Amazon Resource Name (ARN)](https://docs.aws.amazon.com/secretsmanager/latest/userguide/reference_iam-permissions.html#iam-resources) of your secret on the secret details page. 
```

### Create a vector index in your database {#create-vector-index}

> Once your database is set up, you need to create an index with a vector field as your knowledge base for Amazon Bedrock.

Reviewer: ```suggestion
After your database is set up, create an index with a vector field as your knowledge base for Amazon Bedrock.
```

#### [RedisInsight](https://redis.io/docs/ui/insight/)

RedisInsight is a free Redis GUI that allows you to visualize and optimize your data in Redis. 

Follow the steps in [this dedicated guide](https://github.com/RedisVentures/aws-redis-bedrock-stack/blob/main/docs/vector-index-creation.md) to create your vector index with RedisInsight.

#### [`redis-cli`]({{<relref "/rs/references/cli-utilities/redis-cli">}})

> The `redis-cli` command-line utility lets you connect and run Redis commands directly from the command line. To gain access to `redis-cli`, you can [install Redis](https://redis.io/docs/getting-started/).

Reviewer: ```suggestion
The `redis-cli` command-line utility lets you connect and run Redis commands directly from the command line. To use `redis-cli`, you can [install Redis](https://redis.io/docs/getting-started/).
```

Follow the instructions to [connect to your database with TLS certificates]({{<relref "/rc/security/database-security/tls-ssl#connect-with-the-redis-cli">}}).

> 

Reviewer 2: Seems like there's some missing text for the CLI example shown. Maybe something like, "Here's an example of <squidblat>:"
